src/main/scala/com/mwz/sonar/scala/scoverage/ScoverageReportParser.scala

```diff
@@ -29,21 +29,29 @@ import scala.xml.{Node, XML}
 trait ScoverageReportParser extends ScoverageReportParserAPI {
 
   /** Parses the scoverage report from a file and returns the ModuleCoverage. */
-  override def parse(scoverageReportPath: Path, sourcePrefixes: List[Path]): ModuleCoverage = {
+  override def parse(
+    scoverageReportPath: Path,
+    modulePath: Path,
+    sourcePrefixes: List[Path]
+  ): ModuleCoverage = {
     val scoverageXMLReport = XML.loadFile(scoverageReportPath.toFile)
     val moduleScoverage = extractScoverageFromNode(scoverageXMLReport)
 
     val classCoverages = for {
       classNode <- scoverageXMLReport \\ "class"
       scoverageFilename = Paths.get(classNode \@ "filename")
-      filename <- sourcePrefixes.collectFirst {
-        case prefix
-            if PathUtils.cwd
-              .resolve(prefix)
-              .resolve(PathUtils.stripOutPrefix(prefix, scoverageFilename))
-              .toFile
-              .exists =>
-          prefix.resolve(PathUtils.stripOutPrefix(prefix, scoverageFilename)).toString
+      filename <- sourcePrefixes map { prefix =>
```

Comment:
Yes...
I'll comment here a summary of why, just give like 30 minutes I finish something. :+1:

Reply:
Ok, I'll try to include every important detail. The story goes something like this

First, when I made the **ScoverageSensor** refactor (#34). I face a problem, the **Scoverage** reports generated by **`SBT`** had _filenames_ were relative to packages folder. (That means something like **_"mwz/sonar/scala/ScalaPlugin.scala"_**). But the _filenames_ that **SonarQube** returns were relative to the project root folder. That means something like **_"src/main/scala/mwz/sonar/scala/ScalaPlugin.scala"_**).
At that time I naively thought that we only need to add the **_"src/main/scala"_** prefix to all **Scoverage** _filenames_.
But I wasn't happy enough with that, so I asked you if that was a safe assumption, you suggested that we should add a property to define that prefix and default it to **_"src/main/scala"_**. I decided to reuse the property `sonar.sources`, because "I didn't see any value in having another one which will end with the same value." (as you can see [here](https://github.com/mwz/sonar-scala/pull/34#issuecomment-375435025))

Sadly we (mostly me) made bad assumptions:

+ Every **Scoverage** report tool will have the same format.
+ The `sonar.sources` property will have only one value, and it will be the same the user provide.
+ The plugin will work the same with multi-module projects.

So, the problems began. First was #51, there we saw that the **`mvn` SonarQube** plugin changes the `sonar.source` property to be full path and that it can have multiple values. We also noticed that the plugin didn't work in Windows. The solution was to relativize every value in `sonar.source` to the CWD and test if it + the _filename_ exist.

Then was #56 part 1, there the problem was that **`gradle`** generates reports with _filenames_ relative to the project root, so adding the prefix would break the filenames. The solution was to check if the filename already have the prefix or not.

Then in #56 part 2, the problem was that for multi-module projects the `sonar.sources` property includes the module folder thus the prefix removal process would fail. And the solution is this PR.

**Conclusion**
We end up with a very complex process to handle all this, but most of it is caused by inconsistencies of the different tools.
I think we could come with a simpler solution, for example add a property to define the prefix _(what an irony)_.

PS: Sorry for taking so long. Too many things to do, too little time. And sorry in my English was worst than usual, I'm in a hurry.